File: April2022/Java/ImplementStackusingQueues.java
// Author: Shobhit Behl (LC: shobhitbruh)
class MyStack {
    Queue<Integer> q1;
    Queue<Integer> q2;
    public MyStack() {
        q1=new LinkedList<>();
        q2=new LinkedList<>();
    }
    
    public void push(int x) {
        q1.add(x);
    }
    
    public int pop() {
        while(q1.size()>1){
            q2.add(q1.remove());
        }
        int j=q1.remove();
        
        while(q2.size()>0){
            q1.add(q2.remove());
        }
        
        return j;
        
    }
    
    public int top() {
         while(q1.size()>1){
            q2.add(q1.remove());
        }
        int j=q1.remove();
        while(q2.size()>0){
            q1.add(q2.remove());
        }
        q1.add(j);
        
        return j;
    }
    
    public boolean empty() {
       return (q1.size()==0); 
    }
}

File: April2022/Kotlin/ImplementStackusingQueues.kt
// Author: Shobhit Behl (LC: shobhitbruh)
class MyStack() {
    
    var q1: Queue<Int>
    var q2: Queue<Int>
    fun push(x: Int) {
        q1.add(x)
    }

    fun pop(): Int {
        while (q1.size > 1) {
            q2.add(q1.remove())
        }
        val j: Int = q1.remove()
        while (q2.size > 0) {
            q1.add(q2.remove())
        }
        return j
    }

    fun top(): Int {
        while (q1.size > 1) {
            q2.add(q1.remove())
        }
        val j: Int = q1.remove()
        while (q2.size > 0) {
            q1.add(q2.remove())
        }
        q1.add(j)
        return j
    }

    fun empty(): Boolean {
        return q1.size === 0
    }

    init {
        q1 = LinkedList()
        q2 = LinkedList()
    }
}